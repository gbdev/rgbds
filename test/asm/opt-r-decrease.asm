MACRO m
  DEF depth += 1
  IF depth == 10
    OPT r9
  ENDC
  println "at depth {d:depth}"
  m
ENDM

  DEF depth = 0
  m
