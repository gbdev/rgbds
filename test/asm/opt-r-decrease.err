FATAL: opt-r-decrease.asm(11) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(7) -> opt-r-decrease.asm::m(4):
    Recursion limit (9) exceeded
