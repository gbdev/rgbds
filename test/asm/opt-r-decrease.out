at depth 1
at depth 2
at depth 3
at depth 4
at depth 5
at depth 6
at depth 7
at depth 8
at depth 9
